### PR TITLE
wip: looking for feedback

### PR DESCRIPTION
**Pull request by justinmerrell**
As I am working on the RunPod integration, I am migrating the info from the Google Doc to this repo as well as cleaning up some of the examples. 

Still a work in progress, but looking for feedback on what I have thus far. 

For the example files, I am using `FILL_IN` vs `TODO` as the first time I looked at the examples, I misinterpreted "TODO" with incomplete examples that needed work from the managers vs something I was supposed to fill out.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,55 @@
...
+1. [Introductory Blogpost](https://medium.com/@zongheng_yang/skypilot-ml-and-data-science-on-any-cloud-with-massive-cost-savings-244189cc7c0f) [Start here if you are new]
+2. [Documentation](https://skypilot.readthedocs.io/en/latest/)
+3. [The Sky Above the Clouds](https://arxiv.org/abs/2205.07147)
+4. [GitHub](https://github.com/skypilot-org/skypilot)
+
+## How does SkyPilot work?
+
+Here's a simplified overview of SkyPilot's architecture.
+
+In this diagram, the user has two clouds enabled (AWS and GCP). This is what happens when a user launches a job with sky launch:
```
Comment (concretevitamin) [inline README.md:24]:
```suggestion
TODO: diagram in google doc

In this diagram, the user has two clouds enabled (AWS and GCP). This is what happens when a user launches a job with sky launch:
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,55 @@
...
+2. The provisioner+executor executes ray commands to launch a cluster on AWS.
+   - AWS Node Provider is the interface between ray and AWS, translating ray function calls to AWS API calls.
+3. Once the cluster is launched, the provisioner+executor ssh’s into the cluster to execute some AWS Setup commands. This is used to download some important packages on the cluster.
+4. The provisioner+executor submits the job to the cluster and the cluster runs the job.
+
+When all is done, the user can run sky down and provisioner+executor will tear down the cluster by executing more ray commands.
+
+## Getting Started
+
+SkyPilot currently supports five possible clouds (AWS, GCP, Azure, RunPod, and Lambda Cloud). Now let's say you have a new cloud, called FluffyCloud, that you want SkyPilot to support. What do you need to do?
```
Comment (concretevitamin) [inline README.md]:
```suggestion
Now let's say you have a new cloud, called FluffyCloud, that you want SkyPilot to support. What do you need to do?
```
Deleted sentence is not most up to date as docs website

File: docs/integration_steps/step_2-catalog.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+| `GPUInfo`          | string | The human readable information of the GPU (not used in code).                       |
+| `Region`           | string | The region of the resource.                                                         |
+| `AvailabilityZone` | string | The availability zone of the resource (can be empty if not supported in the cloud). |
+| `Price`            | float  | The price of the resource.                                                          |
+| `SpotPrice`        | float  | The spot price of the resource.                                                     |
+
+
+## Parsing Catalog
+
+Create a copy of `fluffycloud.py` and place it at `sky/clouds/service_catalog/{cloudname_catalog}.py` aside from reanaming fluffly cloud to your cloud name, you do not need to make additional changes.
```
Comment (concretevitamin) [inline docs/integration_steps/step_2-catalog.md]:
```suggestion
Create a copy of `fluffycloud_catalog.py` and place it at `sky/clouds/service_catalog/{cloudname_catalog}.py`. Aside from renaming fluffly cloud to your cloud name, you do not need to make additional changes.
```

File: docs/integration_steps/step_5-e2e-failover.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            for s in stderr_splits:
+                print(s)
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError('Errors occurred during provision; '
+                                   'check logs above.')
+
+        logger.warning(f'Got error(s) in {region.name}:')
+        messages = '\n\t'.join(errors)
+        logger.warning(f'{style.DIM}\t{messages}{style.RESET_ALL}')
+        self._blocked_resources.add(launchable_resources.copy(zone=None))
```
Comment (concretevitamin) [inline docs/integration_steps/step_5-e2e-failover.md:36]:
```suggestion
        # NOTE: you can check out other clouds' implementations of this function,
        # which may intelligently block a whole zone / whole region depending on
        # the errors thrown.
        self._blocked_resources.add(launchable_resources.copy(zone=None))
```

File: fluffycloud/fluffycloud.py [hunk 5]
```diff
@@ -267,3 +273,33 @@ def accelerator_in_region_or_zone(self,
                                       zone: Optional[str] = None) -> bool:
         return service_catalog.accelerator_in_region_or_zone(
             accelerator, acc_count, region, zone, 'fluffycloud')
+
+    @classmethod
+    def query_status(cls, name: str, tag_filters: Dict[str, str],
```
Comment (concretevitamin) [inline fluffycloud/fluffycloud.py:278]: Nice!

File: fluffycloud/node_provider.py [hunk 6]
```diff
@@ -1,21 +1,39 @@
+'''
```
Comment (concretevitamin) [inline fluffycloud/node_provider.py]: style nit: the repo uses triple quotes throughout (please edit elsewhere in this module too?)

**Review by concretevitamin — COMMENTED** [review]
Looks great, thanks a bunch for these updates @justinmerrell! Left some minor nits.

File: fluffycloud/node_provider.py (hunk 6 above)
Comment (justinmerrell) [reply fluffycloud/node_provider.py]: Fair enough, done.

**Review by concretevitamin — APPROVED** [review]
Thanks a bunch for this @justinmerrell! Going forward it's going to help new clouds :)

**Pull request merged**